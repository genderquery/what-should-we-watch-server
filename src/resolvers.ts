export default {
  Query: {
    movie: (source: any, args: any, context: any) =>
      context.dataSources.tmdb.getMovieById(args.id),
  },
};
