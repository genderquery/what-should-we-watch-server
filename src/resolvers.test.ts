import resolvers from "./resolvers";

const context = {
  dataSources: {
    tmdb: {
      getMovieById: () => ({
        id: "603",
        title: "The Matrix",
        releaseDate: "1999-03-30",
        overview:
          "Set in the 22nd century, The Matrix tells the story of a computer hacker who joins a group of underground insurgents fighting the vast and powerful computers who now rule the earth.",
      }),
    },
  },
};

test("movie", () => {
  const result = resolvers.Query.movie(null, { id: 603 }, context);
  expect(result).toMatchSnapshot();
});
