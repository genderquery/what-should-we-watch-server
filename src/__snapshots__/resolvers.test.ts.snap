// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movie 1`] = `
Object {
  "id": "603",
  "overview": "Set in the 22nd century, The Matrix tells the story of a computer hacker who joins a group of underground insurgents fighting the vast and powerful computers who now rule the earth.",
  "releaseDate": "1999-03-30",
  "title": "The Matrix",
}
`;
